packages/pluggableWidgets/background-gradient-native/src/BackgroundGradient.tsx

```diff
@@ -13,7 +13,37 @@ export function BackgroundGradient({ name, colorList, content, onClick, style }:
         Alert.alert("", "The widget 'Background gradient' requires an updated 'Make It Native 9' application");
     }
 
+    const opacityValidation = (opacity: number | undefined): number => {
+        if (opacity === undefined) {
+            return defaultStyle.opacity;
+        }
+        const opacityVal = Number(opacity);
+        if (isNaN(opacityVal)) {
+            throw new Error("Opacity must be a number.");
+        }
+        if (opacityVal < 0 || opacityVal > 1) {
+            console.warn("Opacity must be between 0.0 and 1.0");
+        }
+
+        return opacityVal;
+    };
+
+    const angleValidation = (angle: number | undefined): number => {
```

Comment:
the same as the other function, I would move it out of the component.

Reply:
done